src/Symfony/Component/Security/Core/Role/RoleHierarchyInterface.php

```diff
@@ -15,6 +15,8 @@
  * RoleHierarchyInterface is the interface for a role hierarchy.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @deprecated since Symfony 4.3, to be removed in 5.0.
```

Comment:
Why deprecating this interface instead of asking to implement the new method in it ? Removing the interface would remove a supported extension point.

Reply:
I don't remember why I did that initially. I guess it was due to an earlier implementation. #30388 will revert this part.